### PR TITLE
Add pumpkin and sunflower as farmable plants

### PR DESCRIPTION
**Pull request by Rail-Runner**
1. Add pumpkin, its' irradiated version and its' seeds. Also add it to some recipes.
2. Add sunflower seeds. Also add it to cooking oil recipe.
3. Add fried seeds. Can be found along with other snacks or crafted.
4. Add mashed pumpkin. Can be found along with other canned food or crafted.

Other changes:
1. Remove crafting step in harvesting canola seeds: its' seeds are now harvested right away without needing to craft them using a harvested item.
2. Slightly increase amount of product obtained from some plants by removing " "seeds": false " line.

Also, how do I remove those merge commits from the list below in future PRs? They have started to really get in the way.

### CODE REVIEW

**Comment by mutability** [comment]
You can rebase your branch rather than merging from master to avoid the merge spam. This rewrites the history of your branch and you will need to force the push to github as you are no longer just extending the history of your branch, you're replacing it.

**Comment by AppleConnoiseur** [comment]
How could you forget pumpkin pie and pumpkin spice (Can go well in hand with cinnamon) :P? Also many vegetarian recipes can be made with it like butternut pumpkin chili and pumpkin soup.

**Comment by Darkyhard** [comment]
> How could you forget pumpkin pie and pumpkin spice (Can go well in hand with cinnamon) :P? Also many vegetarian recipes can be made with it like butternut pumpkin chili and pumpkin soup.

I didn't add dishes made from pumpkin as separate items (except mashed pumpkin), I've just added pumpkin as a possible component in appopriate recipes.

**Comment by chaosvolt** [comment]
We can always go with adding further uses for these plants in a future PR, yeah. XP

File: data/json/items/comestibles.json [hunk 1]
```diff
@@ -11975,6 +12055,34 @@
...
+        "description" : "This is a simple but tasty dish made by cooking the pumpkin pulp and then mashing.",
+        "price" : 50,
+        "material" : "veggy",
+        "tool" : "null",
+        "volume" : 1,
+        "cutting" : 0,
+        "phase" : "solid",
+        "charges" : 1,
+        "bashing" : 0,
+        "fun" : 5
```
Comment (Coolthulhu) [inline data/json/items/comestibles.json]: 5 fun is quite a lot for something this simple.

Comment (chaosvolt) [inline data/json/items/comestibles.json]: Funny thing I don't think I've ever tried just straight pumpkin, as pumpkin pies are all cinnamon and other things that cause the taste. So not sure what fun value just cooked pumpkin would logically have.

Comment (Rivet-the-Zombie) [inline data/json/items/comestibles.json]: I'd think very little; plain cooked pumpkin is virtually tasteless.

**Comment by chaosvolt** [comment]
Would it be too late for me to add jack-o-lanterns as a follow-up PR? o3o

File: data/json/items/comestibles.json (hunk 1 above)
Comment (Frost-wood) [inline data/json/items/comestibles.json]: Plain cooked pumpkin tastes pretty shitty.  Should be a -1 to fun.

**Comment by Frost-wood** [comment]
Note that all squashes(Pumpkin has the same flavour profile as other squashes) can be made into a good tasting soup easily.  

Ingredients:
1. Butter(or some kind of oil)
2. Squash(Pumpkin, butternut etc).
3. Mirepoix (Onions, Carrots, Celery) 
4. Sugar(Brown, Honey, etc)
5. Stock(White stock, like chicken, veg will work as well)
6. Binder(Flour or similar thickening agent will do.)

Now you can do the soup in two ways, you can either roast all of the ingredients(except the stock), or sweat them in a pot.  When you roast the ingredients add the butter and sugar with the roasted ingredients in a shallow cooking tray.  Cook until the ingredients are soft, put in a pot.

If you added enough oil, add the flour and mix all of the veggies that you cooked together, until it forms a 'paste'.  Then add the stock(or water if you have powdered stock) slowly, mixing it it-adding water too fast will make the soup lumpy, which you could fix with a immersion blender, but being a survivor, you probably don't have access to one that is powered.  Bring to boil, stirring, and cooking until the flour is cooked out.  

Add salt to taste(for salting foods you want add enough to bring foods to life, but not too much so you can taste it).

**Comment by Darkyhard** [comment]
> Note that all squashes(Pumpkin has the same flavour profile as other squashes) can be made into a good tasting soup easily. 

Is it really much diffirent from soups made from other vegetables that it's worth to add a separate item for it? I've added pumpkins as a possible component to vegetable soup recipe already.

**Comment by Rivet-the-Zombie** [comment]
>I've added pumpkins as a possible component to vegetable soup recipe already.

This is perfectly fine. An additional pumpkin soup recipe would be great, but isn't necessary.

File: data/json/items/comestibles.json (hunk 1 above)
Comment (Darkyhard) [inline data/json/items/comestibles.json]: I've changed it to 0. It's still better than eating raw pumpkin.

**Comment by chaosvolt** [comment]
Aye, any new recipes can be puzzled out another day, since existing uses for it are available.

**Comment by Coolthulhu** [comment]
Pumpkin needs balancing. It produces a LOT of nutrition compared to other plants due to its high default charge number. Even at survival 0 you can expect 160 nutrition.

**Comment by Darkyhard** [comment]
> Pumpkin needs balancing. It produces a LOT of nutrition compared to other plants due to its high default charge number. Even at survival 0 you can expect 160 nutrition.

I've noticed it before. The problem is that you can't currently set amount of products harvested from one plant in JSONs, and reducing pumpkins' size (and nutrition) would cause weird stuff: we would have tomato-sized pumpkins or something like this.

I'll attemt to make required code additions to allow to set amount of product harvested, but, as this is not "copy-paste-edit" process, I'm unsure if this will work.

**Comment by Darkyhard** [comment]
Yeah, it works now. Not sure if it actaully works as intended, but at least it doesn't throw any errors.

**Comment by Coolthulhu** [comment]
The charge div seems to work.
Pumpkin is a fair bit too good compared to average plant, but I guess that's OK since it is hard to get seeds for it and carrots and grains still beat it.

**Pull request merged**